addons/mail/static/tests/discuss/core/new_message_separator.test.js

```diff
@@ -0,0 +1,222 @@
+    await click(".o-mail-Composer-send:enabled");
+    await contains(".o-mail-Message", { count: 2 });
+    await contains(".o-mail-Thread-newMessage hr + span", { count: 0, text: "New messages" });
+});
+
+test("remain when switching between chat window and discuss", async () => {
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create({ channel_type: "channel", name: "General" });
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click("button", { text: "General" });
+    await insertText(".o-mail-Composer-input", "Very important message!");
+    await triggerHotkey("Enter");
+    await click(".o-mail-Message [title='Expand']");
+    await click(".o-mail-Message-moreMenu [title='Mark as Unread']");
+    await contains(".o-mail-Thread-newMessage");
+    await click("[title='Open Actions Menu']");
+    await click("[title='Open in Discuss']");
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await contains(".o-mail-Thread-newMessage");
+    await openFormView("res.partner", serverState.partnerId);
+    await contains(".o-mail-ChatWindow-header", { text: "General" });
+    await contains(".o-mail-Thread-newMessage");
+});
+
+test("show on chat upon new message reception (chat window)", async () => {
+    mockDate("2023-01-03 12:00:00"); // so that it's after last interest (mock server is in 2019 by default!)
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Demo" });
+    const userId = pyEnv["res.users"].create({ name: "Foreigner user", partner_id: partnerId });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({
+                fold_state: "open",
+                unpin_dt: "2021-01-01 12:00:00",
+                last_interest_dt: "2021-01-01 10:00:00",
+                partner_id: serverState.partnerId,
+            }),
+            Command.create({ partner_id: partnerId }),
+        ],
+        channel_type: "chat",
+    });
+    const messageId = pyEnv["mail.message"].create({
+        body: "not empty",
+        model: "discuss.channel",
+        res_id: channelId,
+    });
+    const [memberId] = pyEnv["discuss.channel.member"].search([
+        ["channel_id", "=", channelId],
+        ["partner_id", "=", serverState.partnerId],
+    ]);
+    pyEnv["discuss.channel.member"].write([memberId], { seen_message_id: messageId });
+    const env = await start();
+    rpc = rpcWithEnv(env);
+    // simulate receiving a message
+    withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: { body: "hu", message_type: "comment" },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-ChatWindow");
+    await contains(".o-mail-Message", { count: 2 });
+    await contains(".o-mail-Thread-newMessage hr + span", { text: "New messages" });
+});
```

Comment:
This test doesn't make sense with the spec.

When I run it locally, it passes because it shows the new message separator before "not empty" on the initial opening of the chat window. Considering the setup of data in this test: `serverState.partnerId`'s `member.seen_message_id` is on "not empty" message, it should not show the new message separator

Reply:
I'll check it out, I just moved this test.